src/inputs/flow/FlowInputStream.cpp

```diff
@@ -73,7 +73,10 @@ void FlowInputStream::_read_from_pcap_file()
                 sample.rawSampleLen = udpLayer->getLayerPayloadSize();
                 try {
                     read_sflow_datagram(&sample);
-                    sflow_signal(sample);
+                    std::unique_lock lock(_input_mutex);
+                    for (auto &callback : _callbacks) {
+                        dynamic_cast<FlowInputStreamCallback *>(callback.get())->sflow_cb(sample);
```

Comment:
don't like this approach. but I could not use Template with InputStream Class

Reply:
right. well, you can make all of the dynamic_casts static_casts i think, at least. should be faster and we know it will always be this class.